### PR TITLE
hack/run-locally.sh: misc fixes (AWS ovn-kube ports, help text, etc)

### PR DESCRIPTION
**Pull request by dcbw**
1) only open AWS ports when using AWS
2) print help text on errors rather than referring to HACKING.md
3) back up the install-config.yaml so it can be cached by the user

@danwinship @squeed @pecameron @JacobTanenbaum @rcarrillocruz

### CODE REVIEW

File: hack/run-locally.sh [hunk 1]
```diff
@@ -135,6 +135,9 @@ function ensure_install_config() {
     else
         cp "${INSTALL_CONFIG}" "${CLUSTER_DIR}/install-config.yaml"
     fi
+    # since the install-config is consumed by the installer, back it up for
+    # subsequent re-use
+    cp "${CLUSTER_DIR}/install-config.yaml" "${CLUSTER_DIR}/install-config.yaml.bak"
```
Comment (pecameron) [inline hack/run-locally.sh]: you might want to put a sequence number to save more than the last config.

File: hack/run-locally.sh [hunk 2]
```diff
@@ -196,6 +200,7 @@ CLUSTER_DIR="${CLUSTER_DIR:-}"
 INSTALLER_PATH="${INSTALLER_PATH:-}"
 INSTALL_CONFIG="${INSTALL_CONFIG:-}"
 WAIT_FOR_MANIFEST_UPDATES="${WAIT_FOR_MANIFEST_UPDATES:-}"
+PLATFORM=
```
Comment (danwinship) [inline hack/run-locally.sh]: cruft

**Comment by dcbw** [comment]
@danwinship @pecameron PTAL thanks

File: hack/run-locally.sh [hunk 3]
```diff
@@ -180,6 +191,7 @@ The following options are always accepted but only used for new clusters:
  -f               the path to an openshift-install-created install-config.yaml file; if not given one will be created
  -i               path to the openshift-install binary; if not given PATH will be searched
  -n [net plugin]  the name of the network plugin to deploy; one of [sdn|OpenShiftSDN|ovn|OVNKubernetes]
+ -s [secret]      additional pull secret to add to install-config.yaml
```
Comment (danwinship) [inline hack/run-locally.sh]: What's the use case here? Wouldn't it be better to just add that pull secret to your template install-config permanently?

**Comment by dcbw** [comment]
@danwinship removed the "-s" bits and updated the Wiki instructions to combine the secrets outside the installer and just paste them in. PTAL thanks.

**Pull request merged**